src/mon/PGMonitor.cc

```diff
@@ -1310,9 +1310,19 @@ void PGMonitor::_try_mark_pg_stale(
 
 bool PGMonitor::check_down_pgs()
 {
-  dout(10) << "check_down_pgs" << dendl;
+  dout(10) << "check_down_pgs last_osdmap_epoch "
+	   << pg_map.last_osdmap_epoch << dendl;
+  if (pg_map.last_osdmap_epoch == 0)
+    return false;
+
+  // use the OSDMap that matches the one pg_map has consumed.
+  std::unique_ptr<OSDMap> osdmap;
+  bufferlist bl;
+  int err = mon->osdmon()->get_version_full(pg_map.last_osdmap_epoch, bl);
```

Comment:
please ignore the comment above, just read b67ac8c318f8bf635b221b626040b802d2f119c2 , which renders the case above possible